model/notification/notification.go

```diff
@@ -253,3 +253,68 @@ func Find(id bson.ObjectId) (*Notification, error) {
 
 	return &notification, err
 }
+
+type NotificationStats struct {
+	GithubPullRequest int `json:"github_pull_request" bson:"github_pull_request" yaml:"github_pull_request"`
+	JIRAIssue         int `json:"jira_issue" bson:"jira_issue" yaml:"jira_issue"`
+	JIRAComment       int `json:"jira_comment" bson:"jira_comment" yaml:"jira_comment"`
+	EvergreenWebhook  int `json:"evergreen_webhook" bson:"evergreen_webhook" yaml:"evergreen_webhook"`
+	Email             int `json:"email" bson:"email" yaml:"email"`
+	Slack             int `json:"slack" bson:"slack" yaml:"slack"`
+}
+
+func CollectUnsentNotificationStats() (*NotificationStats, error) {
+	const subscriberTypeKey = "type"
+	pipeline := []bson.M{
+		{
+			"$match": bson.M{
+				sentAtKey: bson.M{
+					"$exists": false,
+				},
+			},
+		},
+		{
+			"$group": bson.M{
+				"_id": "$" + bsonutil.GetDottedKeyName(subscriberKey, subscriberTypeKey),
+				"n": bson.M{
+					"$sum": 1,
+				},
+			},
+		},
+	}
+
+	stats := []struct {
+		Key   string `bson:"_id"`
+		Count int    `bson:"n"`
+	}{}
+
+	if err := db.Aggregate(NotificationsCollection, pipeline, &stats); err != nil {
+		return nil, errors.Wrap(err, "failed to count unsent notifications")
+	}
+
+	nStats := NotificationStats{}
+
+	for _, data := range stats {
+		switch data.Key {
+		case event.GithubPullRequestSubscriberType:
+			nStats.GithubPullRequest = data.Count
+
+		case event.JIRAIssueSubscriberType:
+			nStats.JIRAIssue = data.Count
+
+		case event.JIRACommentSubscriberType:
+			nStats.JIRAComment = data.Count
+
+		case event.EvergreenWebhookSubscriberType:
+			nStats.EvergreenWebhook = data.Count
+
+		case event.EmailSubscriberType:
+			nStats.Email = data.Count
+
+		case event.SlackSubscriberType:
+			nStats.Slack = data.Count
+		}
```

Comment:
I think it'd be useful to log/error for an unknown subscriber type

Reply:
done